[Feature][Connector-V2][Iceberg]Modify the scope of flink-shaded-hadoop-2 to provided to be compatible with hadoop3.x

s7monk opened the pull request:
@https://github.com/apache/incubator-seatunnel/issues/3044
…ovided

<!--

Thank you for contributing to SeaTunnel! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/incubator-seatunnel/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

-->

## Purpose of this pull request

<!-- Describe the purpose of this pull request. For example: This pull request adds checkstyle plugin.-->
Modify the scope of flink-shaded-hadoop-2 of the Connector-V2-Iceberg module to provided to be compatible with hadoop3.x
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

s7monk committed:
[Connector-V2-Iceberg]Modify the scope of flink-shaded-hadoop-2 to provided

hailin0 commented on docs/en/connector-v2/sink/Rds.md:
```diff
@@ -0,0 +1,7 @@
+# Rds
```
move to new pull request？

hailin0 commented on docs/en/connector-v2/source/Rds.md:
```diff
@@ -0,0 +1,7 @@
+# Rds
```
move to new pull request？

hailin0 commented on seatunnel-connectors-v2/connector-iceberg/pom.xml:
```diff
@@ -120,6 +120,7 @@
         <dependency>
             <groupId>org.apache.flink</groupId>
             <artifactId>flink-shaded-hadoop-2</artifactId>
+            <scope>provided</scope>
```
update docs

test spark

https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/source/Iceberg.md

s7monk replied:
ok

EricJoy2048 commented:
1. Please add e2e test case in hadoop3.x if you want to support hadoop3.x.

2. You need update https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/connector-v2/source/Iceberg.md and let users know all of hadoop2.x and hadoop3.x are supported.

s7monk commented:
> Please add e2e test case in hadoop3.x

ok

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
[Doc][Connector-V2][Iceberg] Modify Iceberg doc
[Feature][Connector-V2][Iceberg]Modify the scope of hive-exec to provided to be compatible with hadoop3.x

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
Merge branch 'dev' of github.com:s7monk/incubator-seatunnel into dev

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
Merge branch 'dev' of github.com:s7monk/incubator-seatunnel into dev

s7monk committed:
add iceberg connector hadoop3.x e2e

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
Merge branch 'dev' of github.com:s7monk/incubator-seatunnel into dev

s7monk committed:
fix pom error

s7monk committed:
fix iceberg doc error

s7monk commented:
I have tested spark with no problem

s7monk commented:
I have tested the hadoop catalog on hadoop2.7.1 and hadoop3.1.3, and also tested the hive catalog on hive3.1.2, and it runs without problems.

s7monk committed:
fix pom error

s7monk committed:
Merge branch 'apache:dev' into dev

CalvinKirs commented on seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-iceberg-flink-e2e/pom.xml:
```diff
@@ -40,6 +40,17 @@
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-shaded-hadoop-2</artifactId>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.avro</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
```
What is the reason for excluding here? If there will be conflicts, it is better that we provide the corresponding shade jar, otherwise this may hide the problem.

CalvinKirs replied:
cc @hailin0

s7monk replied:
ok

hailin0 commented on seatunnel-e2e/seatunnel-flink-connector-v2-e2e/pom.xml:
```diff
@@ -36,6 +36,7 @@
         <module>connector-fake-flink-e2e</module>
         <module>connector-mongodb-flink-e2e</module>
         <module>connector-iceberg-flink-e2e</module>
+        <module>connector-iceberg-hadoop3-flink-e2e</module>
```
Add module: connector-iceberg-hadoop3-spark-e2e

Or

move module to https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e

s7monk replied:
ok

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
add iceberg haoop3 spark e2e

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
Merge branch 'dev' of github.com:s7monk/incubator-seatunnel into dev

s7monk committed:
add iceberg haoop3 spark e2e

s7monk committed:
add iceberg haoop3 spark e2e

s7monk committed:
add iceberg haoop3 spark e2e

s7monk committed:
Merge branch 'apache:dev' into dev

EricJoy2048 commented:
If this pr is ok, please @ us to review.

s7monk commented:
this pr is  ok , please review it @CalvinKirs @EricJoy2048 @hailin0

s7monk committed:
Merge branch 'dev' into dev

s7monk commented:
PTAL @CalvinKirs @hailin0

Hisoka-X reviewed (dismissed):
LGTM

EricJoy2048 reviewed (dismissed):
LGTM, Thanks for your contribution.

EricJoy2048 committed:
Merge branch 'dev' into dev

hailin0 commented on seatunnel-connectors-v2/connector-iceberg/pom.xml:
```diff
@@ -94,6 +94,7 @@
             <artifactId>hive-exec</artifactId>
             <version>${hive.version}</version>
             <classifier>core</classifier>
+            <scope>provided</scope>
```
Add a note about this into iceberg connector docs?

s7monk replied:
ok

hailin0 commented on seatunnel-connectors-v2/connector-iceberg/pom.xml:
```diff
@@ -120,6 +121,7 @@
         <dependency>
             <groupId>org.apache.flink</groupId>
             <artifactId>flink-shaded-hadoop-2</artifactId>
+            <scope>provided</scope>
```
Add a note about this into iceberg connector docs?

s7monk replied:
ok

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
Merge branch 'dev' into dev

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
modify iceberg doc

EricJoy2048 commented on docs/en/connector-v2/source/Iceberg.md:
```diff
@@ -161,6 +161,19 @@ source {
+
 ## Changelog
 
 ### 2.2.0-beta 2022-09-26
```
Please add the `changed logs`

s7monk replied:
ok

EricJoy2048 replied:
> ok

You can reference other connector.

s7monk replied:
ok

s7monk committed:
Merge branch 'dev' into dev

Commit:
solve some erroe

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
add changed log

s7monk committed:
add changed log

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
solve iceberg test error

s7monk committed:
Merge remote-tracking branch 'origin/dev' into dev

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk committed:
solve iceberg-spark test error

s7monk committed:
Merge remote-tracking branch 'origin/dev' into dev

s7monk committed:
Merge branch 'apache:dev' into dev

s7monk commented:
PTAL @EricJoy2048 @hailin0

Hisoka-X reviewed (approved):
LGTM
